ci/README.md

```diff
@@ -107,6 +117,72 @@ The subsequent PRs on community will be able to pass the
 `check-merge-to-enterprise` check as long as they don't have a conflict with
 `enterprise-master`.
 
+### What to do when your branch got outdated?
+
+So there's one issue that can occur. Your branch will become outdated with
+master and you have to make it up to date. There's two ways to do this using
+`git merge` or `git rebase`. As usual, `git merge` is a bit easier than `git
+rebase`, but clutters git history. This section will explain both. If you don't
+know which one makes the most sense, start with `git rebase`, if for whatever
+reason this doesn't work feel free to fall back to `git merge`.
+
+#### Updating both branches with `git rebase`
+
+In the community repo:
+
+```bash
+git checkout $PR_BRANCH
+git pull origin master --rebase
+git push --force-with-lease
+```
+
+In the enterprise repo:
+
+```bash
+git checkout $PR_BRANCH
+git fetch community
+git rebase community/$PR_BRANCH --preserve-merges
+```
+
+Automatic merge might have fail with the above command. However, because of `git
+rerere` it should have re-applied your original merge resolution. If this is
+indeed the case it should show something like this in the output of the previous
+command (note the `Resolved ...` line):
+```
+CONFLICT (content): Merge conflict in .circleci/config.yml
+Resolved '.circleci/config.yml' using previous resolution.
+Automatic merge failed; fix conflicts and then commit the result.
+Error redoing merge 79a899e5b243696fd1a2754d612ae2eca817cdc1
+```
+
+Confirm that the merge conflict is indeed resolved correctly. In that case you
+can do the following
+```bash
+# Add files that were conflicting
+git add "$(git diff --name-only --diff-filter=U)"
+git rebase --continue
+git push --force-with-lease
+```
+
+#### Updating both branches with `git merge`
+
+In the community repo:
+
+```bash
+git checkout $PR_BRANCH
+git pull origin master
```

Comment:
does this line merge master to current branch or do we need `git merge master` as well? Also it could be nice to mention that in case of conflicts that will need to be fixed to be even more explicit :)

Reply:
made this clearer now, by splitting it in explicit `fetch` and `merge`